src/postgres/src/backend/utils/cache/relcache.c

```diff
@@ -5825,6 +5831,10 @@ RelationCacheInitializePhase3(void)
 		write_relcache_init_file(true);
 		write_relcache_init_file(false);
 	}
+	if (IsYugaByteEnabled() && old_query_id == 0)
+	{
+		YBCSetQueryId(0);
+	}
```

Comment:
Same as above comment, this can incorrectly set the query id to 0.

Reply:
Can you expand on how? 

This code is run only if the existing query-id when entering this function is 0.

So in essence, we are only setting it to 0, if it were 0 when the control entered this function.

Reply:
My bad, I missed line 5834.